frontend/hub/execution-environments/ExecutionEnvironmentForm.tsx

```diff
@@ -0,0 +1,346 @@
+import { useTranslation, TFunction } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import {
+  PageFormSubmitHandler,
+  PageFormTextInput,
+  PageHeader,
+  PageLayout,
+  useGetPageUrl,
+  PageFormTextArea,
+} from '../../../framework';
+import { TagIcon } from '@patternfly/react-icons';
+import { Button, InputGroup, Label, LabelGroup, TextInput } from '@patternfly/react-core';
+import { PageFormGroup } from '../../../framework/PageForm/Inputs/PageFormGroup';
+import { AwxError } from '../../awx/common/AwxError';
+import { useGetRequest } from '../../common/crud/useGet';
+import { HubRoute } from '../HubRoutes';
+import { hubAPI, hubAPIPost, pulpAPI } from '../api/utils';
+import { ExecutionEnvironment } from './ExecutionEnvironment';
+import { HubPageForm } from '../HubPageForm';
+import { HubItemsResponse } from '../useHubView';
+import { useState, useEffect, useCallback } from 'react';
+
+import { patchHubRequest, putHubRequest } from './../api/request';
+import { PageFormAsyncSelect } from '../../../framework/PageForm/Inputs/PageFormAsyncSelect';
+import { useSelectRegistrySingle } from './hooks/useRegistrySelector';
+
+export function ExecutionEnvironmentForm(props: { mode: 'add' | 'edit' }) {
```

Comment:
Isnt that antipattern? You have to maintain some copy paste code?

Reply:
However, I solved the problem of conditional useGet, when you pass different URL, it will do another request and when you pass empty string, it will do nothing. So I was able to rewrite it using mode. If you are comfortable with this solution, it would be totaly ok for me:

I can keep mode under the hood, but I will expose two components Add and Edit form, which will contains component with mode. But outside of form file, there will be no mode.

Reply:
I like the new approach.